blog/2024-11-linked-signal/README.md

```diff
@@ -0,0 +1,337 @@
+The difference lies in their return types: While `computed()` returns a read-only `Signal`, the new `linkedSignal` function creates a `WritableSignal`.
+
+That means, we can override the value using `set()` and `update()` whenever required.
+A signal created with `computed()` does not allow modifying the value manually.
+
+In this first example, we used the **shorthand syntax** for the Linked Signal.
+It is also possible to separate the computation into a second function.
+The value of the source is automatically passed into the computation function.
+
+```ts
+const timestampMs = signal(Date.now());
+
+const timestampSecondsLinked = linkedSignal({
+  source: timestampMs,
+  computation: ms => ms / 1000
+});
+```
+
+Whether or not to use the the more elaborate options object with `source` and `computation` over the simpler shorthand syntax depends on use-case and taste.
+Both examples for `timestampSecondsLinked` above have the exact same behaviour.
+In more complex cases, a separate computation function might make the code more readable.
+
+
+## Basic Usage of Linked Signal
+
+To see how it works, let’s take a look at a complete example. 
+Our component has a list of books in the `books` signal.
+Then we’re using a Linked Signal to keep track of the *first book* in the list, created by the `linkedSignal()` factory function.
+We decided to use the full notation with an options object. The separate computation makes it more readable, compared to a one-line function that combines source and computation.
+
+Whenever the list of books changes, the `firstBook` signal will automatically recalculate its value to the first book in the updated list.
+Up to here, all of this could have been achieved with a Computed Signal.
+However, the Linked Signal makes it possible to manually override the value in the `overrideFirstBook()` method.
+
+```typescript
+import { Component, linkedSignal, signal } from '@angular/core';
+
+@Component({
+  selector: 'app-book-list',
+  template: `
+    <p>First book in list: {{ firstBook() }}</p>
+    <button (click)="changeBookList()">Change Book List</button>`
+})
+export class BookListComponent {
+  books = signal(['Angular', 'React', 'Vue']);
+
+  firstBook = linkedSignal({
+    source: this.books,
+    computation: books => books[0]
+  });
+
+  // this also works (shorthand notation)
+  // firstBook = linkedSignal(() => this.books()[0]);
+
+  overrideFirstBook() {
+    // Manually updating `firstBook`, which now returns 'jQuery'
+    this.firstBook.set('jQuery');
+  }
+
+  changeBookList() {
+    // Changing `books` causes `firstBook` to reset, now returning 'Next.js'
+    this.books.set(['Next.js', 'Svelte', 'Nuxt']);
+  }
+}
+```
+
+In this example:
+- The Linked Signal `firstBook` initially points to the first book in the list of `books`.
+- When `books` changes, `firstBook` recalculates to reflect the first item in the updated list.
+- However, we can also override the value manually, as shown with the 'jQuery' book. 
```

Comment:
What happens when `changeBookList()` is called after `overrideFirstBook()`. Computed again?

Reply:
Yes, the latest value counts. `books` is the source of the Linked Signal, so when it changes, the Signal recomputes.

| Action | `firstBook()` |
| --- | --- |
| init | `Angular` |
| `overrideFirstBook()` | `jQuery` |
| `changeBookList()` | `Next.js` |



| Action | `firstBook()` |
| --- | --- |
| init | `Angular` |
| `changeBookList()` | `Next.js` |
| `overrideFirstBook()` | `jQuery` |